### PR TITLE
Fixing Security Vulternability in Auto-CEST

### PR DESCRIPTION
**Pull request by bastosmichael**
## Description of change
Fixing Security Vulnerability discovered in ticket https://github.com/department-of-veterans-affairs/vets-contrib/issues/1612 with a Santize method

## Testing done
Testing with vulnerability payload

## Testing planned
Adding vulnerability payload to Rspec Tests

## Acceptance Criteria (Definition of Done)
- Needs to Sanitize payload before calling send method

#### Unique to this PR
- [x] Will Sanitize incoming payloads

#### Applies to all PRs

- [x] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: modules/claims_api/app/models/claims_api/form_526.rb [hunk 1]
```diff
@@ -38,23 +40,29 @@ class Form526
     end
 
     def initialize(params = {})
-      @attributes = []
-      params.each do |name, value|
-        @attributes << name.to_sym
-        begin
-          send("#{name}=", value)
-        rescue StandardError
+      sanatized_params = sanatize_fields(params)
```
Comment (patrickvinograd) [inline modules/claims_api/app/models/claims_api/form_526.rb]: sanatized -> sanitized

**Comment by patrickvinograd** [comment]
Agree this fixes the vulnerability, I'm wondering if the metaprogramming using `send` is optimizing for conciseness over clarity. 

Also wondering if defining a JSON Schema for this input and validating against that would be more in line with existing vets-api form submissions.

**Comment by bastosmichael** [comment]
@lihanli Thanks, we actually started using Virtus.model but ran into validation restraints that led us down the path of just using Active Record modules instead.

File: modules/claims_api/app/models/claims_api/form_526.rb (hunk 1 above)
Comment (bastosmichael) [reply modules/claims_api/app/models/claims_api/form_526.rb]: Fixed, sorry about that it was early in the morning lol.

**Comment by kreek** [comment]
Agree with @lihanli about using Virtus you could eliminate using `send` for dynamic setters set in `initialize` as Virtus will map a Hash to the attributes for you. It will also give you `#.to_h` and to `#.to_json`, and it acts as validation in a way as you set up what types you expect.

File: modules/claims_api/app/models/claims_api/form_526.rb [hunk 2]
```diff
@@ -38,23 +40,29 @@ class Form526
...
-          send("#{name}=", value)
-        rescue StandardError
+      sanitized_params = sanitize_fields(params)
+      sanitized_params.each do |name, value|
+        send("#{name}=", value)
+      end
+    end
+
+    def sanitize_fields(params)
+      bad_fields = params.keys.to_a - ALL_FIELDS.map & :to_sym
```
Comment (kreek) [inline modules/claims_api/app/models/claims_api/form_526.rb]: think this needs to be `ALL_FIELDS.map(&:to_sym)`

File: modules/claims_api/app/models/claims_api/form_526.rb [hunk 3]
```diff
@@ -38,23 +40,29 @@ class Form526
...
+      params
     end
 
     def persisted?
       false
     end
 
     def attributes
-      @attributes.map { |method| { method => send(method) } }.reduce(:merge)
+      ALL_FIELDS.map { |method| { method => send(method) } }.reduce(:merge).delete_if { |_k, v| v.blank? }
```
Comment (kreek) [inline modules/claims_api/app/models/claims_api/form_526.rb]: Could we get the same effect of removing keys with empty values by doing [Hash.compact](https://apidock.com/rails/Hash/compact) (or a recursive version of it)?

**Comment by kreek** [comment]
Also agree with @patrickvinograd on conciseness over clarity, and I don't think we're really saving lines of code. The fields are listed in the constants, might as well just set `attr_accessor` (or `attribute` if using virtus) manually at that point. That way a future dev (including you in two months) will not need a few extra minutes to figure out how it works 😄

File: modules/claims_api/app/models/claims_api/form_526.rb (hunk 2 above)
Comment (bastosmichael) [reply modules/claims_api/app/models/claims_api/form_526.rb]: Thanks, made the change.

File: modules/claims_api/app/models/claims_api/form_526.rb (hunk 3 above)
Comment (bastosmichael) [reply modules/claims_api/app/models/claims_api/form_526.rb]: Will consider that for a refactor.

**Comment by bastosmichael** [comment]
Thank you for the feedback @kreek I went ahead and created the ticket for our team to add the strong params next sprint but like I mentioned we went down the Virtus.model route initially and ran into some issues in terms of giving proper error messaging back to the API consumer in order to properly handle validations. After reading up a bit on the creator's thoughts https://www.reddit.com/r/ruby/comments/3sjb24/virtus_to_be_abandoned_by_its_creator/ and how he himself has switched over to dry-validations and dry-data gems I thought it best at least for now to go the Active Model with some minor tweaks.

**Comment by kreek** [comment]
👍sounds good @bastosmichael, dry-validations and dry-data gems work too. I think Virtus might have even been abandoned right when vets-api started. Although 'abandoned' sounds like it's degrading in a dumpster somewhere 😄so far it's yet to give us any issues. I actually might try swapping dry-types in as the author claims a speed increase.

**Pull request merged**